### PR TITLE
Search

### PR DESCRIPTION
**Pull request by nyashafrank**
Search functionality with results popping up as you type. Not sure if there should be "0 results found" if a service doesn't match, or if the user can assume that from not seeing any services in the list. Not sure if the latter is a good idea though because right now, for example, when a user attempts to search "LLC", no matches appear for "L" or "LL". Matches only appear for the full string "LLC". So a user can be typing halfway through a search term and think there are no services for her term since no matches appear so far. Not sure if the fix is in the API or in the front-end code. Thank you for reading my novel.

### CODE REVIEW

**Comment by nyashafrank** [comment]
wait DO NOT pull yet

File: src/reducers/service_reducer.js [hunk 1]
```diff
@@ -1,14 +1,17 @@
...
-import { FETCH_ALL_SERVICES, CREATE_SERVICE } from '../actions';
+import { FETCH_ALL_SERVICES, CREATE_SERVICE, SERVICE_SEARCH } from '../actions';
 
 export default function(state = {}, action) {   
     switch (action.type) {
         case FETCH_ALL_SERVICES: 
             return { services: action.payload.data.services }
         case CREATE_SERVICE:
         console.log(action);
+
```
Comment (abeSanchez) [inline src/reducers/service_reducer.js]: Can you add a return state here and remove the extra empty line? thanks

File: src/actions/index.js [hunk 2]
```diff
@@ -56,6 +56,18 @@ export function fetchServices() {
...
+export function serviceSearch(term){
+    const data = {
+        query: term
+    }
+
+    return {
+        type: SERVICE_SEARCH,
+        payload: axios.post(`${ROOT_URL}/services/search`, data)
+    };
+
```
Comment (abeSanchez) [inline src/actions/index.js]: Can you get rid of this last empty line? thanks

File: src/components/search.js [hunk 3]
```diff
@@ -1,16 +1,58 @@
...
+    constructor(props) {
+
+        // because we extend Component
+        super(props);
+
+        this.state = {term: ''};
+
+        // makes sure we don't lose reference to the function
+        this.onInputChange = this.onInputChange.bind(this);
+    }
```
Comment (abeSanchez) [inline src/components/search.js]: Can you remove these comments and mine at the top too? unless you need them

Comment (abeSanchez) [reply src/components/search.js]: and remove the extra white lines

File: src/reducers/service_reducer.js [hunk 4]
```diff
@@ -1,14 +1,17 @@
...
+        case SERVICE_SEARCH:
+            console.log(action.payload.data.results.length);
```
Comment (abeSanchez) [inline src/reducers/service_reducer.js]: oh also can you remove the logs? thanks

**Pull request merged**